src/dynsight/_internal/analysis/entropy.py

```diff
@@ -153,92 +219,193 @@ def compute_entropy_gain(
 
             import numpy as np
-            from dynsight.analysis import sample_entropy
+            from dynsight.analysis import compute_entropy_gain_multi
 
             np.random.seed(1234)
-            data = np.random.rand(1000)
+            data = np.random.rand(1000, 2)  # 2D dataset
+            n_bins = [40, 40]
+            labels = np.random.randint(-1, 2, size=1000)
 
-            samp_en = sample_entropy(
+            _, entropy_gain, *_ = compute_entropy_gain_multi(
                 data,
-                m_par=2,
-                r_factor=0.2,
+                labels,
+                n_bins=n_bins,
             )
 
-        .. testcode:: sampen1-test
+        .. testcode:: shannon2-multi-test
             :hide:
 
-            assert np.isclose(samp_en, 2.2351853395754424)
+            assert np.isclose(entropy_gain, 0.13171418273750357)
     """
-    n_frames = len(particle)
-    if n_frames < m_par + 1:
-        err_msg = "Time-series too short"
-        raise ValueError(err_msg)
-    r_th = r_factor * np.std(particle)
-
-    # To store counts of similar pairs for m and m+1
-    number_of_pairs = [0, 0]
-
-    for i, m in enumerate([m_par + 1, m_par]):
-        # Create overlapping windows of length m
-        window_list = np.array(
-            [particle[j : j + m] for j in range(n_frames - m + 1)]
+    if data.shape[0] != labels.shape[0]:
+        msg = (
+            f"data ({data.shape}) and labels ({labels.shape}) "
+            "must have same shape[0]"
         )
+        raise RuntimeError(msg)
 
-        # Compute pairwise distances (Chebyshev is typical for SampEn)
-        distances = pdist(window_list, metric="chebyshev")
+    data_range = [(float(np.min(tmp)), float(np.max(tmp))) for tmp in data.T]
 
-        # Count pairs within the threshold r_th
-        number_of_pairs[i] = np.sum(distances < r_th)
+    # Compute the entropy of the raw data
+    total_entropy = compute_shannon_multi(
+        data,
+        data_range,
+        n_bins,
+    )
 
-    if number_of_pairs[1] == 0.0 and number_of_pairs[0] == 0.0:
-        err_msg = "Distance threshold too strict"
-        raise ValueError(err_msg)
+    # Compute the fraction and the entropy of the single clusters
+    n_clusters = np.unique(labels).size
+    frac, entr = np.zeros(n_clusters), np.zeros(n_clusters)
+    for i, label in enumerate(np.unique(labels)):
+        mask = labels == label
+        frac[i] = np.sum(mask) / labels.size
+        entr[i] = compute_shannon_multi(
+            data[mask],
+            data_range,
+            n_bins,
+        )
 
-    return -np.log(number_of_pairs[0] / number_of_pairs[1])
+    # Compute the entropy of the clustered data
+    clustered_entropy = np.dot(frac, entr)
+    info_gain = total_entropy - clustered_entropy
+
+    return (
+        info_gain,
+        info_gain / total_entropy,
+        total_entropy,
+        clustered_entropy,
+    )
+
+
+def pairwise_probabilities(
```

Comment:
This has not been added to the module yet anyone, so it is not publicly available or in the docs